### ISSUE TITLE
Jenkins un-idling not working

### ISSUE THREAD

**Issue by ppitonak** [openshiftio/openshift.io#4215]
##### Issue Overview

Sometimes Jenkins does not un-idle

##### Expected Behaviour

Jenkins work at any time, if it is idled, it would un-idle

##### Current Behaviour

User sees just a message that Jenkins is starting, it doesn't start in 30min.
![05-03-jenkins-direct-log](https://user-images.githubusercontent.com/445100/44339675-8e020180-a482-11e8-85b1-c2398ae64629.png)

##### Steps To Reproduce

1. reset user account
2. create new space and new project (e.g. Vert.X HTTP endpoint)
3. wait for pipeline to finish -> sometimes it doesn't start

##### Additional Information

Originally reported in #3517 by @ldimaggi @ljelinkova 

When the issue #3517 was opened, it was because init container hack in Jenkins.
At the moment

* Init Container hack is removed
* Content repo is removed and jenkins is given more resources

**Comment by ppitonak** [comment]
Still happening with Jenkins 4.0.104

**Comment by sthaha** [comment]
@ppitonak could you please post the following details when you encounter this problem

1. the cluster on which you are facing the issue
2. the pod logs
3. `oc get ev` 
4. if Jenkins pod does get un-idled, could you upload the pod logs as well? 
5. output of `oc get all`

**Comment by ppitonak** [comment]
We don't gather pod logs yet but last failure is here http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-openshift.io-smoketest-us-east-2a-released/550/oc-logs-output.txt

cluster starter-us-east-2a
`oc get ev` empty

**Comment by ljelinkova** [comment]
We have updated the script that retrieves logs from oc and the last result is here:

http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-openshift.io-smoketest-us-east-2a-beta/588/oc-logs-output.txt

**Comment by ppitonak** [comment]
```
NAME                   READY     STATUS             RESTARTS   AGE
pod/jenkins-1-deploy   0/1       DeadlineExceeded   0          4h
```

**Comment by ppitonak** [comment]
Another job failed, status `DeadlineExceeded` is the only useful information that we have
http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-openshift.io-smoketest-us-east-2a-released/610/oc-logs-output.txt

**Comment by ppitonak** [comment]
@sthaha it seems that this bug is more probable on starter-us-east-2a, focus on this cluster.

**Comment by sthaha** [comment]
@ppitonak thanks for uploading the logs. From the logs, it appears like an issue with openshift or the namespace. 
could you please upload the result of `oc get -o yaml dc/jenkins` ?
Is the env reset before running the test?

**Comment by ppitonak** [comment]
@sthaha I'll update our script to gather also these info. We do reset env but it happens approximately 4 hours before next run which should be plenty of time.

**Comment by ljelinkova** [comment]
This is the result of updated script (with oc get -o yaml dc/jenkins command)

http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-openshift.io-smoketest-us-east-1a-beta/628/oc-logs-output.txt

**Comment by ljelinkova** [comment]
This issue still occurs - is there any progress here?

**Comment by sthaha** [comment]
I am going to add more logs to idler to understand why `user-idler` part idles/ unidles pods as a part of solving this bug. Once we have the logs we should have better insights into the "decision" making process of idler which at the moment is hard to figure out without logs.

**Comment by joshuawilson** [comment]
I've also added this to a story to track the e2e bugs at https://openshift.io/openshiftio/Openshift_io/plan/detail/802

**Comment by sthaha** [comment]
@ppitonak we have recently deployed a change to idler to give us a lot more insights into this issue. Could you please post logs / ping us when you hit this issue again? 
Also would it be possible to get a logs of `oc get ev` of `-jenkins` namespace as soon as the test starts?

**Comment by sthaha** [comment]
@ppitonak since you marked it `P1` could you please let us know how frequent do you see this issue?

**Comment by ljelinkova** [comment]
This is prod-preview job with the un-idling error:

https://ci.centos.org/job/devtools-test-e2e-prod-preview.openshift.io-smoketest-pr-us-east-2a-beta/2923/

**Comment by ppitonak** [comment]
Stats from prod-preview: 30 jobs, 21 failures, 10 failures on this issue, i.e. 1 in 3 run fail on this issue, this is nr. 1 issue for prod-preview.

**Comment by ppitonak** [comment]
> > Also would it be possible to get a logs of `oc get ev` of `-jenkins` namespace as soon as the test starts?
> 
> Yes, we can do that but it will take some time

https://github.com/fabric8io/fabric8-test/pull/1069 by @ljelinkova merged, all jobs starting in 15min will gather these events

**Comment by skryzhny** [comment]
Idler fix was deployed to prod 2 days ago and fixed at least gluster/filesystem resource usage problem, did it fix the problem?  
Can someone recheck?

**Comment by joshuawilson** [comment]
@ljelinkova does this issue effect user in production or just the e2e? Severity describes the effects on the user.

**Comment by sthaha** [comment]
@ldimaggi could you try resetting twice and wait a while (10mins or so) ?

**Comment by ppitonak** [comment]
@sthaha I tried multiple resets and long wait on my accounts with no result, basically all my, Lucia's and Jenkins accounts are broken at the moment.

**Comment by sthaha** [comment]
@ljelinkova why has the severity being bumped to `sev-1` ?

**Comment by ppitonak** [comment]
See my previous comment... since we aren't able to get our accounts to usable state, we assume that accounts belonging to external users cannot self-heal and there is no workaround.

**Comment by sthaha** [comment]
Could you please publish the `oc get -w ev` when you go https://jenkins.openshift.io ?

**Comment by ldimaggi** [comment]
@sthaha 

Seeing the PV issue after resets:

10:41:39 AM | jenkins-1-5t6lf | Pod | Warning | Failed Scheduling | persistentvolumeclaim "jenkins-home" not found20 times in the last 8 minutes
10:35:42 AM | jenkins-1-5t6lf | Pod | Warning | Failed Scheduling | persistentvolumeclaim "jenkins-home" is being deleted6 times in the last 9 minutes
10:35:27 AM | jenkins-1 | Replication Controller | Normal | Successful Create | Created pod: jenkins-1-5t6lf

I am on starter-us-east-2, is this problem also being seen elsewhere?

**Comment by ppitonak** [comment]
@sthaha latest results

1. account reset 46 minutes before test start
2. [events before test started](http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-prod-preview.openshift.io-smoketest-pr-us-east-2a-beta/3097/oc-jenkins-logs-before-all.txt)
3. [events after 10 minutes after build started](http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-prod-preview.openshift.io-smoketest-pr-us-east-2a-beta/3097/oc-jenkins-logs.txt)

**Comment by skryzhny** [comment]
I tried with fresh account on 1b and got this:
```
$ oc get  ev --sort-by='.lastTimestamp'
LAST SEEN   FIRST SEEN   COUNT     NAME                                KIND                    SUBOBJECT                     TYPE      REASON                        SOURCE                                   MESSAGE
25m         25m          1         jenkins.155ebce53ceaa40a            DeploymentConfig                                      Normal    DeploymentCreated             deploymentconfig-controller              Created new replication controller "jenkins-1" for version 1
25m         25m          1         jenkins-1-deploy.155ebce548d7506f   Pod                                                   Normal    SuccessfulMountVolume         kubelet, ip-172-22-53-97.ec2.internal    MountVolume.SetUp succeeded for volume "deployer-token-fdzcl" 
25m         25m          1         jenkins-1-deploy.155ebce53fc168ea   Pod                                                   Normal    Scheduled                     default-scheduler                        Successfully assigned jenkins-1-deploy to ip-172-22-53-97.ec2.internal
25m         25m          1         jenkins-1-deploy.155ebce5c9b76d71   Pod                     spec.containers{deployment}   Normal    Pulling                       kubelet, ip-172-22-53-97.ec2.internal    pulling image "registry.reg-aws.openshift.com:443/openshift3/ose-deployer:v3.10.9"
25m         25m          1         jenkins-1-deploy.155ebce5d7765c20   Pod                     spec.containers{deployment}   Normal    Pulled                        kubelet, ip-172-22-53-97.ec2.internal    Successfully pulled image "registry.reg-aws.openshift.com:443/openshift3/ose-deployer:v3.10.9"
25m         25m          1         jenkins-1-deploy.155ebce5ebcd7b0f   Pod                     spec.containers{deployment}   Normal    Started                       kubelet, ip-172-22-53-97.ec2.internal    Started container
25m         25m          1         jenkins-1.155ebce5f9d6705c          ReplicationController                                 Normal    SuccessfulCreate              replication-controller                   Created pod: jenkins-1-lz7nk
25m         25m          1         jenkins-1-lz7nk.155ebce5fa557bdd    Pod                                                   Normal    Scheduled                     default-scheduler                        Successfully assigned jenkins-1-lz7nk to ip-172-22-51-134.ec2.internal
25m         25m          1         jenkins-1-lz7nk.155ebce608e7c115    Pod                                                   Normal    SuccessfulMountVolume         kubelet, ip-172-22-51-134.ec2.internal   MountVolume.SetUp succeeded for volume "jenkins-config" 
25m         25m          1         jenkins-1-lz7nk.155ebce6090ba89c    Pod                                                   Normal    SuccessfulMountVolume         kubelet, ip-172-22-51-134.ec2.internal   MountVolume.SetUp succeeded for volume "jenkins-token-l94r2" 
25m         25m          1         jenkins-1-deploy.155ebce5dbd93162   Pod                     spec.containers{deployment}   Normal    Created                       kubelet, ip-172-22-53-97.ec2.internal    Created container
25m         25m          1         jenkins-1-lz7nk.155ebce616a6270f    Pod                                                   Normal    SuccessfulMountVolume         kubelet, ip-172-22-51-134.ec2.internal   MountVolume.SetUp succeeded for volume "b287e7b2-8802-415f-b80b-1cbc1064a30b-02-f1" 
25m         25m          1         jenkins-1-lz7nk.155ebce69cd4f7d6    Pod                     spec.containers{jenkins}      Normal    Pulling                       kubelet, ip-172-22-51-134.ec2.internal   pulling image "fabric8/jenkins-openshift:v7e8de2d"
25m         25m          1         jenkins-1-lz7nk.155ebce6f7e52e5a    Pod                     spec.containers{jenkins}      Normal    Pulled                        kubelet, ip-172-22-51-134.ec2.internal   Successfully pulled image "fabric8/jenkins-openshift:v7e8de2d"
25m         25m          1         jenkins-1-lz7nk.155ebce6fbe3e381    Pod                     spec.containers{jenkins}      Normal    Created                       kubelet, ip-172-22-51-134.ec2.internal   Created container
25m         25m          1         jenkins-1-lz7nk.155ebce707d05659    Pod                     spec.containers{jenkins}      Normal    Started                       kubelet, ip-172-22-51-134.ec2.internal   Started container
22m         24m          9         jenkins-1-lz7nk.155ebcf2d78cdf1f    Pod                     spec.containers{jenkins}      Warning   Unhealthy                     kubelet, ip-172-22-51-134.ec2.internal   Readiness probe failed: HTTP probe failed with statuscode: 503
22m         22m          1         jenkins-1.155ebd120d68c0ee          ReplicationController                                 Normal    SuccessfulDelete              replication-controller                   Deleted pod: jenkins-1-lz7nk
22m         22m          1         jenkins-1-deploy.155ebd120d42a84f   Pod                     spec.containers{deployment}   Normal    Killing                       kubelet, ip-172-22-53-97.ec2.internal    Killing container with id docker://deployment:Need to kill Pod
22m         22m          1         jenkins.155ebd120cc449cc            DeploymentConfig                                      Normal    ReplicationControllerScaled   deploymentconfig-controller              Scaled replication controller "jenkins-1" from 1 to 0
22m         22m          1         jenkins-1-lz7nk.155ebd1270b2d7ed    Pod                     spec.containers{jenkins}      Normal    Killing                       kubelet, ip-172-22-51-134.ec2.internal   Killing container with id docker://jenkins:Need to kill Pod
```

**Comment by joshuawilson** [comment]
Is this really a `SEV-1`?

**Comment by skryzhny** [comment]
I tried 2 accounts - old on starter-2 and fresh on starter-1b, both have build failed.

**Comment by sthaha** [comment]
I see that the current idler is a bit too aggressive in idling jenkins. i.e it idles as soon as a build is over. I will deploy a fix for this into prod-preview so that it can be tested in prod before deploying to Prod.

**Comment by sthaha** [comment]
This has been fixed and deployed to prod. Please reopen if you experience the same issue happen. 
FWIW, I see that there are intermittent e2e test failure which I think are unrelated to jenkins idler.

**Comment by ppitonak** [comment]
@sthaha unfortunately it still doesn't work

**Comment by sthaha** [comment]
@ppitonak unfortunated "doesn't work" isn't that helpful 


what doesn't work? 

- Did the jenkins not get idled or unidled? 
- Did the job run?
- Did the job get terminated in between 
- Do you have the logs?
- what is user-id you used to test?

**Comment by ppitonak** [comment]
Unfortunately I don't have data, it failed locally with my personal account. I'll close now and reoples when I have some data.

**Comment by ljelinkova** [comment]
The e2e tests failed for user osio-ci-e2e-002 because Jenkins was not unidled.

Jenkins build started but the View link was never displayed. When the e2e navigated to Jenkins page directly it showed idled Jenkins screen. 

These are logs when tests started
http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-openshift.io-smoketest-us-east-1a-beta/1292/oc-jenkins-logs-before-all.txt

These are logs after they failed
http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-openshift.io-smoketest-us-east-1a-beta/1292/oc-jenkins-logs.txt

Both logs contain similar entry 
Before
`pod/jenkins-1-deploy   0/1       DeadlineExceeded   0          1h`
After
`pod/jenkins-1-deploy   0/1       DeadlineExceeded   0          2h`

**Comment by sthaha** [comment]
@ljelinkova I  am going to close this bug. Please open a new one against sd-team if jenkins fails to come up on 1a. 

I see that idler has tried to un-idler jenkins and it seems jenkins failed to wake up due to pvc issues. Idler's role is to `idle` or `unidle` jenkins which it has done. 

```
1h          1h           3         jenkins-1-deploy.156294d9cccd837e   Pod                                                   Normal    DeadlineExceeded              kubelet, ip-172-21-52-39.ec2.internal   Pod was active on the node longer than the specified deadline
1h          1h           1         jenkins-1-deploy.156294d9d4c16173   Pod                     spec.containers{deployment}   Normal    Killing                       kubelet, ip-172-21-52-39.ec2.internal   Killing container with id docker://deployment:Need to kill Pod
1h          2h           191       jenkins-1-jpnrs.156291943c5d234c    Pod                                                   Warning   FailedScheduling              default-scheduler                       persistentvolumeclaim "jenkins-home" not found
1h          1h           1         jenkins-1.156294d9d5f73c90          ReplicationController                                 Normal    SuccessfulDelete              replication-controller                  Deleted pod: jenkins-1-jpnrs
```

**Comment by sthaha** [comment]
@ljelinkova .. I should be fairly straight forward to test this. You don't need idler for testing  if this is an openshift issue. You should be able to `oc idle dc/jenkins` and `oc scale --replicas=1 dc/jenkins` to test if it is an openshift issue.

**Comment by ppitonak** [comment]
@sthaha users don't have rights to idle/unidle by default.

**Comment by sthaha** [comment]
@ppitonak I see the point you are making but idler can't fix openshift if pods don't come up. 

we can propagate the errors back to jenkins proxy ui but idler and proxy can't do anything more than that and this doesn't help when jenkins needs to be woken up in response to a webhook. 

I certainly see your point that there must be some way to report and fix broken namespaces but that is outside the scope of idler and proxy.

**Comment by ljelinkova** [comment]
New issue created to fix infrastructure problem that prevents Jenkins to start

https://github.com/openshiftio/openshift.io/issues/4514
